contrib/annoy-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+option(ENABLE_ANNOY "Enable Annoy index support" ${ENABLE_LIBRARIES})
+
+if (NOT ENABLE_ANNOY)
+    message (STATUS "Not using annoy")
+    return()
+endif()
+
+set(ANNOY_PROJECT_DIR "${ClickHouse_SOURCE_DIR}/contrib/annoy")
+set(ANNOY_SOURCE_DIR "${ANNOY_PROJECT_DIR}/src")
+
+add_library(_annoy ${ANNOY_SOURCE_DIR}/annoylib.h)
```

Comment:
I remember that for another 3rd party library (base-x) which is also header-only, the linker complained that the (binary) library is too small (< 4 kB) and failed. This could be avoided by changing the library into an interface library [0]. This is also the recommended way in CMake to deal with header-only libraries. Perhaps you like to try that out?

[0] http://mariobadr.com/creating-a-header-only-library-with-cmake.html

Reply:
Yepp, just that happened in https://s3.amazonaws.com/clickhouse-builds/37215/656ca5388c8d82c74af1c06966eaeb240885306f/binary_darwin_aarch64/build_log.log

```
Jul 19 07:22:16 ld: file too small (length=8) file 'contrib/annoy-cmake/lib_annoy.a' for architecture arm64
Jul 19 07:22:16 clang: error: linker command failed with exit code 1 (use -v to see invocation)
Jul 19 07:22:16 [6430/6445] Linking CXX executable programs/clickhouse
Jul 19 07:22:16 FAILED: programs/clickhouse 
```

--> needs to be made into a header-only library